sources/HUBContainerView.m

```diff
@@ -25,18 +25,28 @@
 
 @implementation HUBContainerView
 
-- (void)setBackgroundColor:(nullable UIColor *)backgroundColor
+- (void)setCollectionView:(nullable UICollectionView *)collectionView
 {
-    [super setBackgroundColor:backgroundColor];
-    
-    for (UIView * const view in self.subviews) {
-        view.backgroundColor = backgroundColor;
+    if (collectionView == nil) {
```

Comment:
You can avoid some code duplication here by removing the previous collection view before adding the new one (regardless of if the new one is `nil`).